app/assets/javascripts/listings/ListingDataService.js.coffee

```diff
@@ -0,0 +1,434 @@
+  Service.loadListing = (listing) ->
+    return if Service.listing && Service.listing.Id == listing.Id
+    # TODO: won't be needed if we ever consolidate Listing_Lottery_Preferences and /preferences API
+    listing.preferences = _.map listing.Listing_Lottery_Preferences, (lotteryPref) ->
+      {
+        listingPreferenceID: lotteryPref.Id
+        preferenceName: lotteryPref.Lottery_Preference.Name
+      }
+    angular.copy(listing, Service.listing)
+
+  Service.seniorMinimumAge = (listing) ->
+    if listing.Reserved_community_minimum_age
+      "#{listing.Reserved_community_minimum_age}+"
+    else
+      ''
+
+  Service.formattedAddress = (listing, type='Building', display='full') ->
+    street = "#{type}_Street_Address"
+    zip = "#{type}_Postal_Code"
+    if type == 'Leasing_Agent'
+      street = "#{type}_Street"
+      zip = "#{type}_Zip"
+    else if type == 'Building'
+      zip = "#{type}_Zip_Code"
+
+    # If Street address is undefined, then return false for display and google map lookup
+    if listing[street] == undefined
+      return
+    # If other fields are undefined, proceed, with special string formatting
+    if listing[street] != undefined
+      Street_Address = listing[street] + ', '
+    else
+      Street_Address = ''
+    if listing["#{type}_City"] != undefined
+      City = listing["#{type}_City"]
+    else
+      City = ''
+    if listing["#{type}_State"] != undefined
+      State = listing["#{type}_State"]
+    else
+      State = ''
+    if listing[zip] != undefined
+      Zip_Code = listing[zip]
+    else
+      Zip_Code = ''
+
+    if display == 'street'
+      return "#{Street_Address}"
+    else if display == 'city-state-zip'
+      return "#{City} #{State}, #{Zip_Code}"
+    else
+      "#{Street_Address}#{City} #{State}, #{Zip_Code}"
+
+  Service.getListingDownloadURLs = ->
+    urls = angular.copy(ListingConstantsService.defaultApplicationURLs)
+    english = _.find(urls, { language: 'English' })
+    chinese = _.find(urls, { language: 'Traditional Chinese' })
+    spanish = _.find(urls, { language: 'Spanish' })
+    tagalog = _.find(urls, { language: 'Tagalog' })
+    # replace download URLs if they are customized on the listing
+    listing = Service.listing
+    english.url = listing.Download_URL if listing.Download_URL
+    chinese.url = listing.Download_URL_Cantonese if listing.Download_URL_Cantonese
+    spanish.url = listing.Download_URL_Spanish if listing.Download_URL_Spanish
+    tagalog.url = listing.Download_URL_Tagalog if listing.Download_URL_Tagalog
+    angular.copy(urls, Service.listingDownloadURLs)
+
+  Service.getProjectIdForBoundaryMatching = (listing) ->
+    return unless listing
+    if ListingPreferenceService.hasPreference('antiDisplacement', listing)
+      'ADHP'
+    else if ListingPreferenceService.hasPreference('neighborhoodResidence', listing)
+      listing.Project_ID
+    else
+      null
+
+  Service.mapSlugToId = (id) ->
```

Comment:
looks like this is only used as part of the hardcoding scheme. maybe we can add a TODO to this function that says "TODO: Remove when we remove hardcoded listings

Reply:
i've put a note to do that in the story about removing hardcoding in general: https://www.pivotaltracker.com/story/show/163463841